File: app/src/main/java/include/mvp/mainscreen/MainActivity.java
package include.mvp.mainscreen;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import include.mvp.App;
import include.mvp.R;
import include.mvp.data.component.DaggerMainScreenComponent;
import include.mvp.data.module.MainScreenModule;
import include.mvp.model.Post;
import include.mvp.presenter.MainPresenter;

public class MainActivity extends BaseActivity implements MainScreenContract.View{

    ListView listView;
    ArrayList<String> list;
    ArrayAdapter<String> adapter;

    @Inject
    MainPresenter mainPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listView = (ListView) findViewById(R.id.my_list);
        list = new ArrayList<>();

        DaggerMainScreenComponent.builder()
                .netComponent(((App) getApplicationContext()).getNetComponent())
                .mainScreenModule(new MainScreenModule(this))
                .build().inject(this);

        mainPresenter.loadPost();

    }

    @Override
    public void showPosts(List<Post> posts) {
        //Loop through the posts and get the title of the post and add it to our list object
        for (int i = 0; i < posts.size(); i++) {
            list.add(posts.get(i).getTitle());
        }
        //Create the array adapter and set it to list view
        adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, list);
        listView.setAdapter(adapter);
    }

    @Override
    public void showError(String message) {
        Toast.makeText(getApplicationContext(), "Error" + message, Toast.LENGTH_SHORT).show();

    }

    @Override
    public void showComplete() {
        Toast.makeText(getApplicationContext(), "Complete", Toast.LENGTH_SHORT).show();

    }
}

File: app/src/main/java/include/mvp/mainscreen/MainActivity.kt
package include.mvp.mainscreen

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast

import java.util.ArrayList

import javax.inject.Inject

import butterknife.BindView
import butterknife.ButterKnife
import include.mvp.R
import include.mvp.model.Post
import include.mvp.presenter.MainPresenter

class MainActivity : BaseActivity(), MainScreenContract.View {

    @BindView(R.id.my_list)
    lateinit var listView: ListView

    @Inject
    lateinit var mainPresenter: MainPresenter

    private var list: ArrayList<String> = ArrayList()
    private var adapter: ArrayAdapter<String>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ButterKnife.bind(this)

        applicationComponent.inject(this)
        mainPresenter.bind(this)
        mainPresenter.loadPost()

    }

    override fun onDestroy() {
        mainPresenter.unbind()
        super.onDestroy()
    }

    override fun showPosts(posts: List<Post>) {
        for (i in posts.indices) {
            posts[i].title?.let { list.add(it) }
        }

        adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, list)
        listView.adapter = adapter
    }

    override fun showError(message: String) {
        Toast.makeText(applicationContext, "Error" + message, Toast.LENGTH_SHORT).show()

    }

    override fun showComplete() {
        Toast.makeText(applicationContext, "Complete", Toast.LENGTH_SHORT).show()

    }
}
